File: csharp/InsecureCrypto.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace SastSamples
{
    public static class InsecureCrypto
    {
        public static void Main()
        {
            Console.Write("Password: ");
            var password = Console.ReadLine()!;

            // CWE-327: MD5 is cryptographically broken.
            using var md5 = MD5.Create();
            var hash = md5.ComputeHash(Encoding.UTF8.GetBytes(password));
            Console.WriteLine("MD5: " + Convert.ToHexString(hash));
        }
    }
}

File: java/InsecureCrypto.java
import java.security.MessageDigest;

public class InsecureCrypto {
    public static void main(String[] args) throws Exception {
        if (args.length == 0) return;
        // CWE-327: MD5 is cryptographically broken.
        MessageDigest md5 = MessageDigest.getInstance("MD5");
        byte[] digest = md5.digest(args[0].getBytes());
        StringBuilder sb = new StringBuilder();
        for (byte b : digest) sb.append(String.format("%02x", b));
        System.out.println("MD5: " + sb);
    }
}
